Sources/BuildSystemIntegration/BuildSystemHooks.swift

```diff
@@ -31,12 +31,6 @@ package struct SwiftPMTestHooks: Sendable {
   }
 }
 
-/// When running SourceKit-LSP in-process, allows the creator of `SourceKitLSPServer` to create the build system instead
-/// of SourceKit-LSP creating build systems as needed.
-package protocol BuildSystemInjector: Sendable {
```

Comment:
I think I mildly prefer the protocol over passing a loose closure, but don't feel strongly. Alternatively you could make it a struct that wraps the closure to avoid repeating the type

Reply:
I prefer the closure-based approach 😉